src/viam/sdk/components/camera/camera.cpp

```diff
@@ -24,6 +25,96 @@ API API::traits<Camera>::api() {
     return {kRDK, kComponent, "camera"};
 }
 
+// Appends a uint64_t value in big-endian format to a byte vector and updates the offset.
+void append_uint64_big_endian(std::vector<unsigned char>& data, size_t& offset, uint64_t value) {
+    const uint64_t value_be = boost::endian::native_to_big(value);
+    if (data.size() < offset + 8) {
+        throw Exception("Incorrect data size-- attempted to write beyond data bounds");
```

Comment:
`size--` looks like a C++ expression here. I'd write this as `"Incorrect data size: attempted ..."`